File: old/Assembly-CSharp_V1/SRPG/JobRankParam.cs
// Decompiled with JetBrains decompiler
// Type: SRPG.JobRankParam
// Assembly: Assembly-CSharp, Version=1.2.0.0, Culture=neutral, PublicKeyToken=null
// MVID: 9BA76916-D0BD-4DB6-A90B-FE0BCC53E511
// Assembly location: C:\Users\André\Desktop\Assembly-CSharp.dll

namespace SRPG
{
  public class JobRankParam
  {
    public static readonly int MAX_RANKUP_EQUIPS = 6;
    public string[] JobChangeItems = new string[3];
    public int[] JobChangeItemNums = new int[3];
    public string[] equips = new string[JobRankParam.MAX_RANKUP_EQUIPS];
    public StatusParam status = new StatusParam();
    public BaseStatus TransfarStatus = new BaseStatus();
    public OInt avoid = (OInt) 0;
    public OInt inimp = (OInt) 0;
    public int JobChangeCost;
    public int cost;
    public OString[] learnings;

    public bool Deserialize(JSON_JobRankParam json)
    {
      if (json == null)
        return false;
      this.JobChangeCost = json.chcost;
      this.JobChangeItems[0] = json.chitm1;
      this.JobChangeItems[1] = json.chitm2;
      this.JobChangeItems[2] = json.chitm3;
      this.JobChangeItemNums[0] = json.chnum1;
      this.JobChangeItemNums[1] = json.chnum2;
      this.JobChangeItemNums[2] = json.chnum3;
      this.cost = json.cost;
      this.equips[0] = json.eqid1;
      this.equips[1] = json.eqid2;
      this.equips[2] = json.eqid3;
      this.equips[3] = json.eqid4;
      this.equips[4] = json.eqid5;
      this.equips[5] = json.eqid6;
      this.status.hp = (OInt) json.hp;
      this.status.mp = (OShort) json.mp;
      this.status.atk = (OShort) json.atk;
      this.status.def = (OShort) json.def;
      this.status.mag = (OShort) json.mag;
      this.status.mnd = (OShort) json.mnd;
      this.status.dex = (OShort) json.dex;
      this.status.spd = (OShort) json.spd;
      this.status.cri = (OShort) json.cri;
      this.status.luk = (OShort) json.luk;
      this.avoid = (OInt) json.avoid;
      this.inimp = (OInt) json.inimp;
      this.learnings = (OString[]) null;
      int length = 0;
      if (!string.IsNullOrEmpty(json.learn1))
        ++length;
      if (!string.IsNullOrEmpty(json.learn2))
        ++length;
      if (!string.IsNullOrEmpty(json.learn3))
        ++length;
      if (length > 0)
      {
        this.learnings = new OString[length];
        int num1 = 0;
        if (!string.IsNullOrEmpty(json.learn1))
          this.learnings[num1++] = (OString) json.learn1;
        if (!string.IsNullOrEmpty(json.learn2))
          this.learnings[num1++] = (OString) json.learn2;
        if (!string.IsNullOrEmpty(json.learn3))
        {
          OString[] learnings = this.learnings;
          int index = num1;
          int num2 = index + 1;
          learnings[index] = (OString) json.learn3;
        }
      }
      return true;
    }
  }
}

File: Database_V2/ParamFunctions/JobRankParam.py
def JobRankParam(json):
    this={}#JobRankParamjson)
    if 'chcost' in json:
        this['JobChangeCost'] = json['chcost']
    this['JobChangeItems']=[None]*3
    if 'chitm1' in json:
        this['JobChangeItems'][0] = json['chitm1']
    if 'chitm2' in json:
        this['JobChangeItems'][1] = json['chitm2']
    if 'chitm3' in json:
        this['JobChangeItems'][2] = json['chitm3']
    this['JobChangeItemsNums']=[None]*3
    if 'chnum1' in json:
        this['JobChangeItemNums'][0] = json['chnum1']
    if 'chnum2' in json:
        this['JobChangeItemNums'][1] = json['chnum2']
    if 'chnum3' in json:
        this['JobChangeItemNums'][2] = json['chnum3']
    if 'cost' in json:
        this['cost'] = json['cost']
    this['equips']=[None]*6
    if 'eqid1' in json:
        this['equips'][0] = json['eqid1']
    if 'eqid2' in json:
        this['equips'][1] = json['eqid2']
    if 'eqid3' in json:
        this['equips'][2] = json['eqid3']
    if 'eqid4' in json:
        this['equips'][3] = json['eqid4']
    if 'eqid5' in json:
        this['equips'][4] = json['eqid5']
    if 'eqid6' in json:
        this['equips'][5] = json['eqid6']

    this['status']={}
    if 'hp' in json:
        this['status']['hp'] = json['hp']
    this['status']
    if 'mp' in json:
        this['status']['mp'] = json['mp']
    this['status']
    if 'atk' in json:
        this['status']['atk'] = json['atk']
    this['status']
    if 'def' in json:
        this['status']['def'] = json['def']
    this['status']
    if 'mag' in json:
        this['status']['mag'] = json['mag']
    this['status']
    if 'mnd' in json:
        this['status']['mnd'] = json['mnd']
    this['status']
    if 'dex' in json:
        this['status']['dex'] = json['dex']
    this['status']
    if 'spd' in json:
        this['status']['spd'] = json['spd']
    this['status']
    if 'cri' in json:
        this['status']['cri'] = json['cri']
    this['status']
    if 'luk' in json:
        this['status']['luk'] = json['luk']
    if 'avoid' in json:
        this['avoid'] = json['avoid']
    if 'inimp' in json:
        this['inimp'] = json['inimp']

    this['learnings']=[]
    for i in range(1,4):
        learn='learn'+str(i)
        if learn in json:
            this['learnings'].append(json[learn])
    if len(this['learnings'])==0:
        del this['learnings']

    return this
